WooCommerce/Classes/AppDelegate.swift

```diff
@@ -578,6 +577,44 @@ private extension AppDelegate {
     }
 }
 
+// MARK: - Magic link
+private extension AppDelegate {
+    func handleAuthenticationUrl(_ url: URL, options: [UIApplication.OpenURLOptionsKey: Any], rootViewController: UIViewController) -> Bool {
+        return if ServiceLocator.stores.isAuthenticatedWithoutWPCom {
+            handleAuthenticationUrlForJetpackSetup(url, rootViewController: rootViewController)
+        } else {
+            ServiceLocator.authenticationManager.handleAuthenticationUrl(url, options: options, rootViewController: rootViewController)
+        }
+    }
+
+    func handleAuthenticationUrlForJetpackSetup(_ url: URL, rootViewController: UIViewController) -> Bool {
+        if let site = ServiceLocator.stores.sessionManager.defaultSite {
+            return handleAuthenticationUrlForJetpackSetup(with: site, url: url, rootViewController: rootViewController)
+        } else {
+            // Wait for the site to be available before handling the magic link
+            ServiceLocator.stores.sessionManager.defaultSitePublisher
+                .timeout(.seconds(30), scheduler: DispatchQueue.main)
+                .first(where: { $0 != nil })
+                .sink { site in
+                    guard let site = site else {
+                        return
```

Comment:
❓ Should we log an error here to know when handling the deeplink fails due to missing site?

Reply:
Gotcha, `site` should not be optional here since we already filtered the non-nil value in the `first` operator above. Please leave a comment stating that if you don't mind.